File: 0001_TwoSum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            Integer index = map.get(target - nums[i]);
            if (index != null) {
                result[0] = index;
                result[1] = i;
                break;
            } else {
                map.put(nums[i], i);
            }
        }
        return result;
    }
}

File: 0001.Two Sum.go
func towSum(nums []int, target int) []int {
    result := make([]int, 2)
    tmpMap := make(map[int][int])

    length := len(nums)
    for i := 0; i < length; i++ {
        value, exist = tmpMap[target - nums[i]]
        if exist {
            result[0] = value
            result[1] = i
            break
        } else {
            tmpMap[nums[i]] = i
        }
    }

    return result
}